PSReadLine/History.cs

```diff
@@ -76,6 +78,11 @@ public class HistoryItem
         private const string _failedForwardISearchPrompt = "failed-fwd-i-search: ";
         private const string _failedBackwardISearchPrompt = "failed-bck-i-search: ";
 
+        // Pattern used to check for sensitive inputs.
+        private static Regex s_sensitivePattern = new Regex(
+            "password|asplaintext|token|key|secret",
```

Comment:
My history doesn't contain any secrets (I'm careful because I'm aware of the problem), and I use most of these terms, so this would unnecessarily impact my history (verified

I tested this regex against my history and had 147 hits - none contained secrets.

At a minimum, one should be able to remove or customize the regex.

But I also wonder if instead we should be using the regular expression(s) the CredScan uses.

Reply:
I don't think credscan is OSS.  Intent here is to start with something and we can improve upon it.  This regex was an example from @LeeHolmes on what he uses.  I do agree we should have an Option to opt out of this.

Reply:
I agree. If we want to expose this option, I think it should be a handler just like the `AddToHistoryHandler`, maybe call it `DetectSensitiveInputHandler`.
By the way, it looks we don't handle the exception thrown from the custom `AddToHistoryHandler`.

Reply:
In some ways, education might be more valuable. After all, `bash` has the same problem.

This change is silent. What about a different change that warns instead? Warning when reading the history file might be a good option too, with a way to disable it of course.

Also - I'm following up on the possibility of using something from CredScan.

Reply:
Oh, that's nice. Curious if we can borrow something simple from CredScan.

Reply:
Two options are added:
```
Set-PSReadLineOption
    [-ScrubSensitiveHistory]
    [-DetectSensitiveInputHandler <Func[String, Boolean]>]
```
By default, `ScrubSensitiveHistory` is set to `true`. When it's set to `false`, all history commands will be saved to the history file.
`DetectSensitiveInputHandler` is used only when `ScrubSensitiveHistory` is enabled, to allow a user to define what is considered to be sensitive.
When it's null, the default logic will be used, which is the regex pattern borrowed from Lee.

Reply:
I don't think we'll get anything from CredScan - they worry people will easily bypass the checks intentionally if the tests are shared.

Reply:
That's a very reasonable concern :)